packages/shared/src/react/hooks/useReverification.ts

```diff
@@ -74,16 +103,24 @@ function createReverificationHandler(params: { onOpenModal: Clerk['__experimenta
   return assertReverification;
 }
 
-function __experimental_useReverification<Fetcher extends () => Promise<any>>(fetcher: Fetcher): readonly [Fetcher] {
+function __experimental_useReverification<
+  Fetcher extends (...args: any[]) => Promise<any>,
+  O extends UseReverificationOptions,
+>(
+  fetcher: Fetcher,
+  options?: O,
+): readonly [(...args: Parameters<Fetcher>) => Promise<ExcludeClerkError<Awaited<ReturnType<Fetcher>>, O>>] {
```

Comment:
👀  let's use more readable names:

```suggestion
  Options extends UseReverificationOptions,
>(
  fetcher: Fetcher,
  options?: Options,
): readonly [(...args: Parameters<Fetcher>) => Promise<ExcludeClerkError<Awaited<ReturnType<Fetcher>>, Options>>] {
```